jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/DefaultCredentialRetrievers.java

```diff
@@ -166,29 +166,33 @@ public DefaultCredentialRetrievers setCredentialHelper(@Nullable String credenti
       credentialRetrievers.add(inferredCredentialRetriever);
     }
 
+    String dockerConfigEnv = environment.get("DOCKER_CONFIG");
+    if (dockerConfigEnv != null) {
+      addDockerFiles(credentialRetrievers, Paths.get(dockerConfigEnv));
+    }
+
     String homeProperty = systemProperties.getProperty("user.home");
-    String homeEnvVar = environment.get("HOME");
     if (homeProperty != null) {
```

Comment:
Do you think it's worth avoiding duplicates by checking if `Paths.get(homeProperty).resolve(DOCKER_DIRECTORY)` is same as `Paths.get(dockerConfigEnv)`? Or not? We've been comparing `homeEnvVar` and `homeProperty`. I just want some consistency for future readers of the code who might assume there is a reason for the inconsistency.

Reply:
I think we can do it. It adds a little extra complexity but it's still straightforward enough.